openmp/runtime/src/kmp_tasking.cpp

```diff
@@ -5454,25 +5447,86 @@ bool __kmpc_omp_has_task_team(kmp_int32 gtid) {
 
 #if OMPX_TASKGRAPH
 // __kmp_find_tdg: identify a TDG through its ID
-// gtid:   Global Thread ID
 // tdg_id: ID of the TDG
 // returns: If a TDG corresponding to this ID is found and not
 // its initial state, return the pointer to it, otherwise nullptr
 static kmp_tdg_info_t *__kmp_find_tdg(kmp_int32 tdg_id) {
   kmp_tdg_info_t *res = nullptr;
+  kmp_int32 i = 0;
   if (__kmp_max_tdgs == 0)
     return res;
 
   if (__kmp_global_tdgs == NULL)
     __kmp_global_tdgs = (kmp_tdg_info_t **)__kmp_allocate(
         sizeof(kmp_tdg_info_t *) * __kmp_max_tdgs);
 
-  if ((__kmp_global_tdgs[tdg_id]) &&
-      (__kmp_global_tdgs[tdg_id]->tdg_status != KMP_TDG_NONE))
-    res = __kmp_global_tdgs[tdg_id];
+  for (i = 0; i < __kmp_max_tdgs; i++) {
```

Comment:
[optional] name the iteration variable something easier to understand (e.g., tgd_id)

Reply:
updated var name to `tdg_idx`